app/helpers/application_helper/button/smart_state_scan.rb

```diff
@@ -0,0 +1,19 @@
+class ApplicationHelper::Button::SmartStateScan < ApplicationHelper::Button::Basic
+  needs :@record
+
+  SMART_ROLES = %w(SmartProxy SmartState).freeze
+
+  def check_smart_roles
+    my_zone = MiqServer.my_server.my_zone
+    SMART_ROLES.each do |role|
+      unless MiqServer.all.any? { |s| s.has_active_role?(role.downcase) && (s.my_zone == my_zone) }
```

Comment:
@enoodle if this is the only place where you're using `SMART_ROLES` why don't you keep them `downcase`'d in the constant?